common/errno_mapping.cc

```diff
@@ -166,4 +166,12 @@ absl::Status ErrnoToCanonicalStatus(int error_number,
                       absl::StrCat(message, ": ", strerror(error_number)));
 }
 
+absl::Status ErrorCodeToCanonicalStatus(const std::error_code& code,
+                                        absl::string_view message) {
+  absl::StatusCode absl_code = code.category() == std::system_category()
```

Comment:
Please return no error in case the code is zero:

`if (!code) return absl::OkStatus();`

Reply:
Done.